core/src/main/java/org/springframework/security/authentication/DefaultAuthenticationEventPublisher.java

```diff
@@ -160,6 +160,25 @@ public void setAdditionalExceptionMappings(Properties additionalExceptionMapping
 		}
 	}
 
+	/**
+	 * Overload of {@link #setAdditionalExceptionMappings(Properties)}}, providing the ability
+	 * to map authentication exceptions and failure authentication events using a {@link Map}
+	 *
+	 * @param mappings where keys are exception classes and values are event classes.
+	 */
+	public void setAdditionalExceptionMappings(Map<Class<? extends AuthenticationException>,
+			Class<? extends AbstractAuthenticationFailureEvent>> mappings){
+		Assert.notEmpty(mappings, "The mappings Map must not be empty nor null");
+		for (Map.Entry<Class<? extends AuthenticationException>,Class<? extends AbstractAuthenticationFailureEvent>> entry
+				: mappings.entrySet()) {
+				Class<?> exceptionClazz = entry.getKey();
+				Class<?> eventClazz = entry.getValue();
+				Assert.isAssignable(AuthenticationException.class, exceptionClazz);
```

Comment:
Is there a reason you are checking `isAssignable`? It seems like the parameter types for the `Map` are doing this for you. I think if someone bypasses the parameter types through erasure or something, they do so at their own peril.

What would you think of something more like:

```java
Class<? extends AuthenticationException> exceptionClass = entry.getKey();
Assert.notNull(exceptionClass, "exceptionClass cannot be null");
```